### PR TITLE
add sequence number support to NL, don't double send data from AvatarMixer

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 305835ea79506b940bf4b787e169361c6ea2ad5f available at: https://gist.github.com/1206c0193b50cc0857f5<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4795.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4795.dmg

File: assignment-client/src/avatars/AvatarMixer.cpp [hunk 1]
```diff
@@ -257,8 +254,37 @@ void AvatarMixer::broadcastAvatarData() {
                         && distribution(generator) > (nodeData->getFullRateDistance() / distanceToAvatar)) {
                         return;
                     }
+
+                    PacketSequenceNumber lastSeqToReceiver = nodeData->getLastBroadcastSequenceNumber(otherNode->getUUID());
+                    PacketSequenceNumber lastSeqFromSender = otherNode->getLastSequenceNumberForPacketType(PacketTypeAvatarData);
+
+                    assert(lastSeqToReceiver <= lastSeqFromSender);
```
Comment (Atlante45) [inline assignment-client/src/avatars/AvatarMixer.cpp]: If you move that inside the if below, you'll test only once. #MicroOptimisation

**Comment by ZappoMan** [comment]
since we have sequence numbers already in a lot of the packets (octree data, octree edits, audio, etc)... what is the plan for migrating all packets to using this same technique?

**Comment by birarda** [comment]
@ZappoMan the plan is to slowly migrate them across where appropriate - didn't feel like that was required for this initial pull

**Comment by ZappoMan** [comment]
I wasn't saying it's required for this pull... but I'm just pointing out that we've got more protocols using/needing sequence numbers than not.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 805250d1e1ec21ec84d51be0271447f857f9559b available at: https://gist.github.com/1206c0193b50cc0857f5<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4795.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-4795.dmg

**Pull request merged**